cmd/main.go

```diff
@@ -461,12 +459,27 @@ func main() {
 		}
 	}()
 
-	address := fmt.Sprintf(":%s", swag.StringValue(port))
 	if Options.ServeHTTPS {
-		log.Fatal(http.ListenAndServeTLS(address, Options.HTTPSCertFile, Options.HTTPSKeyFile, h))
-	} else {
-		log.Fatal(http.ListenAndServe(address, h))
+		log.Infof("Serving https on port %d", Options.HTTPSPort)
+		address := fmt.Sprintf(":%d", Options.HTTPSPort)
+		go func() {
+			httpsHandler := h
+			log.Fatal(http.ListenAndServeTLS(address, Options.HTTPSCertFile, Options.HTTPSKeyFile, httpsHandler))
+		}()
+
+		// If we're serving https, we only want to allow downloading the iso over http so add the middleware here
+		allowList := []*app.RequestMatch{
+			{
+				Path:    regexp.MustCompile(`/clusters/[\w-]+/downloads/image$`),
+				Methods: []string{http.MethodGet, http.MethodHead},
+			},
+		}
+		h = app.WithPathAllowListMiddleware(h, allowList)
```

Comment:
Do we have a way to test this? Even if it's a simple unit-test for now

Reply:
I was going to look into this today before removing the hold :+1:

Reply:
I added tests specifically for this method in the `app` package.

One of them covers this case specifically. Let me know what you think, otherwise I could extract this case somewhere and test that.